types/bluebird/bluebird-tests.ts

```diff
@@ -241,43 +242,46 @@ barProm = barProm.then((value: Bar) => {
 
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 
-fooProm = fooProm.catch((reason: any) => {
+// $ExpectType Bluebird<void | Foo>
+fooProm.catch((reason: any) => {
 	return;
 });
 
-fooProm = fooProm.caught((reason: any) => {
+// $ExpectType Bluebird<void | Foo>
+fooProm.caught((reason: any) => {
 	return;
 });
-fooProm = fooProm.catch((error: any) => {
+// $ExpectType Bluebird<void | Foo>
+fooProm.catch((error: any) => {
 	return true;
 }, (reason: any) => {
 	return;
 });
-fooProm = fooProm.caught((error: any) => {
+// $ExpectType Bluebird<void | Foo>
+fooProm.caught((error: any) => {
 	return true;
 }, (reason: any) => {
 	return;
 });
 
-fooProm = fooProm.catch((reason: any) => {
+fooOrVoidProm = fooProm.catch((reason: any) => {
```

Comment:
// $ExpectType Bluebird<void | Foo>
Expect Error! "Bluebird<Foo>" not assign "Bluebird<void | Foo>"